This patch is a port of the upstream commit missing from latest oot irdma
release (1.12.55):

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=b002760f877c0d91ecd3c78565b52f4bbac379dd

---
 src/irdma/uk.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/irdma/uk.c b/src/irdma/uk.c
index 13b9006..dcd98db 100644
--- a/src/irdma/uk.c
+++ b/src/irdma/uk.c
@@ -99,16 +99,16 @@ static int irdma_nop_1(struct irdma_qp_uk *qp)
  */
 void irdma_clr_wqes(struct irdma_qp_uk *qp, u32 qp_wqe_idx)
 {
-	__le64 *wqe;
+	struct irdma_qp_quanta *sq;
 	u32 wqe_idx;
 
 	if (!(qp_wqe_idx & 0x7F)) {
 		wqe_idx = (qp_wqe_idx + 128) % qp->sq_ring.size;
-		wqe = qp->sq_base[wqe_idx].elem;
+		sq = qp->sq_base + wqe_idx;
 		if (wqe_idx)
-			memset(wqe, qp->swqe_polarity ? 0 : 0xFF, 0x1000);
+			memset(sq, qp->swqe_polarity ? 0 : 0xFF, 128 * sizeof(*sq));
 		else
-			memset(wqe, qp->swqe_polarity ? 0xFF : 0, 0x1000);
+			memset(sq, qp->swqe_polarity ? 0xFF : 0, 128 * sizeof(*sq));
 	}
 }
 
-- 

